l2-frontend/src/construct/ConstructPrice.vue

```diff
@@ -179,45 +253,116 @@ export default {
+      return this.coastResearches.filter(coastResearch => {
         const research = coastResearch.research.title.toLowerCase();
         const searchTerm = this.search.toLowerCase();
 
         return research.includes(searchTerm);
       });
     },
+    priceIsActive() {
+      return this.activeStatus.ok;
+    },
+    priceIsSelected() {
+      return !!this.selectedPrice;
+    },
   },
   watch: {
     selectedPrice() {
-      this.getCurrentCoastResearchesInPrice();
+      if (!this.selectedPrice) {
+        this.priceData = {
+          title: '',
+          start: '',
+          end: '',
+          company: null,
+        };
+      } else {
+        this.getCoastsResearchesInPrice();
+        this.checkPriceHidden();
+        this.getPriceData();
+      }
     },
   },
   mounted() {
-    this.getPriceList();
+    this.getPrices();
     this.getResearchList();
+    this.getCompanies();
   },
   methods: {
-    async getPriceList() {
-      this.priceList = await this.$api('/get-price-list');
+    async getPrices() {
+      this.prices = await this.$api('/get-prices');
+    },
+    async updatePrice() {
+      if (this.priceIsSelected) {
+        const { ok, message } = await this.$api('update-price', {
+          id: this.selectedPrice,
+          title: this.priceData.title,
+          start: this.priceData.start,
+          end: this.priceData.end,
+          company: this.priceData.company,
+        });
+        if (ok) {
+          this.$root.$emit('msg', 'ok', 'Прайс обновлен');
+          await this.getPrices();
+        } else {
+          this.$root.$emit('msg', 'error', message);
+        }
+      } else {
+        const { ok, message } = await this.$api('update-price', {
+          id: -1,
+          title: this.priceData.title,
+          start: this.priceData.start,
+          end: this.priceData.end,
+          company: this.priceData.company,
+        });
+        if (ok) {
+          this.$root.$emit('msg', 'ok', 'Прайс добавлен');
+          await this.getPrices();
+        } else {
+          this.$root.$emit('msg', 'error', message);
+        }
+      }
+    },
+    async checkPriceHidden() {
+      if (this.selectedPrice) {
+        this.activeStatus = await this.$api('/check-price-active', this.selectedPrice);
+      }
+    },
+    async getPriceData() {
+      const price = await this.$api('/get-price-data', this.selectedPrice);
+      this.priceData = price.data;
     },
     async getResearchList() {
       this.researchList = await this.$api('/get-research-list');
     },
-    async getCurrentCoastResearchesInPrice() {
-      this.coastResearches = await this.$api('/get-current-coast-researches-in-price', this.selectedPrice);
-      this.originalCoastResearch = this.coastResearches.data;
+    async getCompanies() {
+      this.companies = await this.$api('/get-companies');
+    },
+    async getCoastsResearchesInPrice() {
+      const coast = await this.$api('/get-coasts-researches-in-price', this.selectedPrice);
```

Comment:
Странно напрямую прокидывать значение.
Лучше 
```suggestion
      const coast = await this.$api('/get-coasts-researches-in-price', {id: this.selectedPrice});
```